packages/db/src/repositories/competition.ts

```diff
@@ -1777,44 +1780,23 @@ export class CompetitionRepository {
         `getAgentRankingsInCompetitions called for agent ${agentId} in ${competitionIds.length} competitions`,
       );
 
-      // Calculate rankings directly in SQL without fetching all agents
-      // Only include active agents in ranking calculation
-      const rankingResults = await this.#db.execute(
-        sql`
-        WITH latest_snapshots AS (
-          SELECT
-            ps.competition_id,
-            ps.agent_id,
-            ps.total_value,
-            ROW_NUMBER() OVER (PARTITION BY ps.competition_id, ps.agent_id ORDER BY ps.timestamp DESC) as rn
-          FROM ${portfolioSnapshots} ps
-          INNER JOIN ${competitionAgents} ca ON ps.agent_id = ca.agent_id AND ps.competition_id = ca.competition_id
-          WHERE ps.competition_id IN (${sql.join(
-            competitionIds.map((id) => sql`${id}`),
-            sql`, `,
-          )})
-            AND ca.status = ${"active"}
-        ),
-        ranked AS (
-          SELECT
-            competition_id,
-            agent_id,
-            total_value,
-            RANK() OVER (PARTITION BY competition_id ORDER BY total_value DESC) as rank,
-            COUNT(*) OVER (PARTITION BY competition_id) as total_agents
-          FROM latest_snapshots
-          WHERE rn = 1
-        )
-        SELECT
-          competition_id,
-          rank,
-          total_agents
-        FROM ranked
-        WHERE agent_id = ${agentId}
-      `,
-      );
+      // Get competition statuses to determine which approach to use
+      const competitionStatuses = await this.#db
```

Comment:
this fixes a bug where the "placement" deviates from the actual competition outcome. from what i can gather, the portfolio snapshots table might take snapshot _after_ a comp ends...so it can't fully be trusted. this is especially true for legacy comps.

instead, for any ended comp, we just get the cached value from the `competitions_leaderboard` table.

see: https://app.recall.network/agents/c654e391-c942-4191-903d-8ce7a592fb55

<img width="1374" height="152" alt="Image" src="https://github.com/user-attachments/assets/c044cb7e-3eb1-4e45-b233-1660ab474e93" />

<img width="1352" height="264" alt="Image" src="https://github.com/user-attachments/assets/cf4a0e5d-f8ec-4d48-9794-0d198ad0f394" />